OpenRA.Mods.Common/Traits/BotModules/SupportPowerBotModule.cs

```diff
@@ -151,11 +153,26 @@ void IBotTick.BotTick(IBot bot)
 
 					var frozenTargets = player.FrozenActorLayer != null ? player.FrozenActorLayer.FrozenActorsInRegion(region) : Enumerable.Empty<FrozenActor>();
 					var consideredAttractiveness = powerDecision.GetAttractiveness(targets, player) + powerDecision.GetAttractiveness(frozenTargets, player);
-					if (consideredAttractiveness <= bestAttractiveness || consideredAttractiveness < powerDecision.MinimumAttractiveness)
+					if (consideredAttractiveness < powerDecision.MinimumAttractiveness)
 						continue;
 
-					bestAttractiveness = consideredAttractiveness;
-					bestLocation = new MPos(i, j).ToCPos(map);
+					suitableLocations.Add(new int[] { i, j, consideredAttractiveness });
+					totalAttractiveness += consideredAttractiveness;
+				}
+			}
+
+			if (suitableLocations.Any())
+			{
+				// Only consider locations of above average attractiveness.
+				var avgAttractiveness = totalAttractiveness / suitableLocations.Count();
```

Comment:
General comments:
* avoid arbitrarily shortening words like average to avg.
* when you have a `List` you should always call `.Count` without the `()` to avoid unnecessary LINQ allocations.

Reply:
Specific comment: this whole block should be able to simplify to something like:
```csharp
if (suitableLocations.Count == 0)
	return null;

// Pick a random location with above average attractiveness.
var averageAttractiveness = totalAttractiveness / suitableLocations.Count;
return suitableLocations.Shuffle(world.LocalRandom)
	.First(x => x.Attractiveness >= averageAttractiveness)
	.UV.ToCPos(map);
```

but note that I haven't tested this ingame.

Reply:
It looks like similar simplifications can be applied to `FindFineAttackLocationToSupportPower` too. Would you mind doing that while you're here?